components/workspace/hooks.tsx

```diff
@@ -95,6 +98,77 @@ export const useWorkspaceEmptyMutation = () => {
   return mutation
 }
 
+export const useWorkspaceExportMutation = () => {
+  const labelerAgent = useLabelerAgent()
+  return useMutation({
+    mutationFn: async (items: string[]) => {
+      // Items are exported in groups so we can expect all items in the group to be of same type
+      // For now, only support exporting accounts
+      if (!items[0].startsWith('did:')) {
+        toast.error(`Exporting is only enabled for accounts.`)
+        return []
+      }
+
+      const data: Record<
+        string,
+        ToolsOzoneModerationDefs.RepoViewDetail | null
+      > = {}
+
+      for (const itemChunk of chunkArray(items, 50)) {
+        await Promise.all(
+          itemChunk.map(async (did) => {
+            try {
+              const { data: repo } =
+                await labelerAgent.tools.ozone.moderation.getRepo({
+                  did,
+                })
+
+              data[did] = repo
+            } catch (error) {
+              // For now we're just swallowing errors and exporting what I can
+              console.error(error)
+              data[did] = null
+            }
+          }),
+        )
+      }
+
+      downloadCSV(
+        createCSV({
+          headers: [
+            'did',
+            'handle',
+            'email',
+            'ip',
+            'name',
+            'labels',
+            'profile',
+          ],
+          lines: Object.values(data).map((repo) => {
+            if (!repo) return ''
+            const profile = AppBskyActorProfile.isRecord(repo.relatedRecords[0])
+              ? (repo.relatedRecords[0] as AppBskyActorProfile.Record)
+              : null
+
+            const line: string[] = [
+              repo.did,
+              repo.handle,
+              repo.email || 'Unknown',
+              `${repo.ip || 'Unknown'}`,
+              `${profile?.displayName || 'Unknown'}`,
+              repo.labels?.map(({ val }) => val).join(', ') || 'None',
```

Comment:
I almost wonder if an empty value could be more useful than a default value here.  I suppose it depends how the CSV will be used.

Reply:
yeah I think our current use case is for mod folks to get these exports and review manually in which case, I feel like the explicit default values are more helpful.